src/analysis/analysis.py

```diff
@@ -138,6 +124,35 @@ def main(data_filepath, out_filepath):
         column='Model:N'
     ).save(os.path.join(out_filepath,'model_performance.png'))
 
+def train_random_forest(X_train, y_train, preprocessor, scoring_metrics, cross_val_results):
+    pipe_rf = make_pipeline(preprocessor, RandomForestClassifier(class_weight='balanced', random_state=123))
+    cross_val_results['random forest'] = pd.DataFrame(cross_validate(
+        pipe_rf, X_train, y_train, n_jobs=-1, scoring=scoring_metrics)).agg(['mean']).round(3).T
+
+def train_svc(X_train, y_train, preprocessor, scoring_metrics, cross_val_results):
+    pipe_svc = make_pipeline(preprocessor, SVC(class_weight='balanced', random_state=123))
+    cross_val_results['svc'] = pd.DataFrame(cross_validate(
+        pipe_svc, X_train, y_train, n_jobs=-1, scoring=scoring_metrics)).agg(['mean']).round(3).T
+    cross_val_results['svc']
+
+def train_nb(X_train, y_train, preprocessor, scoring_metrics, cross_val_results):
+    pipe_nb = make_pipeline(preprocessor, BernoulliNB(alpha = 0.1))
+    cross_val_results['bayes'] = pd.DataFrame(cross_validate(
+        pipe_nb, X_train, y_train, n_jobs=-1, scoring=scoring_metrics)).agg(['mean']).round(3).T
+    cross_val_results['bayes']
+
+def train_logreg(X_train, y_train, preprocessor, scoring_metrics, cross_val_results):
+    pipe_lr = make_pipeline(preprocessor, LogisticRegression(max_iter=10000,random_state=123, class_weight='balanced'))
+    cross_val_results['logreg'] = pd.DataFrame(cross_validate(
+        pipe_lr, X_train, y_train, n_jobs=-1, scoring=scoring_metrics)).agg(['mean']).round(3).T
+    cross_val_results['logreg']
+
+def train_dummy(X_train, y_train, preprocessor, scoring_metrics, cross_val_results):
```

Comment:
It's unclear that these functions modify the `cross_val_results` object in place and don't return anything. Perhaps something more explicit like either returning the cross_val_results object at the end or likely better is to return the results dataframe instead of assigning it in these functions.

So instead making the function calls like:
```suggestion
# in the main function
cross_val_results["svc"] = train_svc(X_train, y_train, preprocessor, scoring_metrics)
```

Reply:
done